jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/snmp/SnmpSession.java

```diff
@@ -133,9 +134,17 @@ public ResponseEvent set(PDU pdu, Snmp snmpob, UserTarget targetob, Boolean keep
       // Try to send the SNMP request (synchronously)
       ResponseEvent responseEvent = null;
       try {
-         responseEvent = snmpob.set(pdu, targetob);
-         if (!keepOpen) {
-            snmpob.close();
+         byte[] authEngineID = snmpob.discoverAuthoritativeEngineID(targetob.getAddress(), 1000);
```

Comment:
What is this auth engine ID and where does it come from?

Reply:
The auth engine ID is the ID of the snmp agent you're sending to (in this case the RSU). This discover method sends a get request to the RSU which then returns its auth engine ID. Without an auth engine ID the snmp4j library will just send an empty PDU.